lib/RateLimiterDynamo.js

```diff
@@ -0,0 +1,422 @@
+
+  /**
+   * Initializes a new instance of the class.
+   *
+   * @param {number} RCU - the read capacity units
+   * @param {number} WCU - the write capacity units
+   */
+  constructor(RCU, WCU) {
+    this.readCapacityUnits = RCU;
+    this.writeCapacityUnits = WCU;
+  }
+}
+
+/**
+ * Implementation of RateLimiterStoreAbstract using DynamoDB.
+ * @class RateLimiterDynamo
+ * @extends RateLimiterStoreAbstract
+ */
+class RateLimiterDynamo extends RateLimiterStoreAbstract {
+
+    /**
+     * Constructs a new instance of the class.
+     * The storeClient MUST be an instance of AWS.DynamoDB NOT of AWS.DynamoDBClient.
+     *
+     * @param {Object} opts - The options for the constructor.
+     * @param {function} cb - The callback function (optional).
+     * @return {void}
+     */
+    constructor(opts, cb = null) {
+        super(opts);
+
+        this.client = opts.storeClient;
+        this.tableName = opts.tableName;
+        this.tableCreated = opts.tableCreated;
+        
+        if (!this.tableCreated) {
+          this._createTable(opts.dynamoTableOpts)
+          .then((data) => {
+            this.tableCreated = true;
+
+            this._setTTL()
+            .finally(() => {
+              // Callback invocation
+              if (typeof cb === 'function') {
+                cb();
+              }
+            });
+            
+          })
+          .catch( err => {
+            //callback invocation
+            if (typeof cb === 'function') {
+              cb(err);
+            } else {
+              throw err;
+            }
+          });
+
+        } else {
+
+          this._setTTL()
+          .finally(() => {
+            // Callback invocation
+            if (typeof cb === 'function') {
+              cb();
+            }
+          });
+        }
+    }
+
+    get tableName() {
+        return this._tableName;
+    }
+
+    set tableName(value) {
+        this._tableName = typeof value === 'undefined' ? 'node-rate-limiter-flexible' : value;
+    }
+
+    get tableCreated() {
+        return this._tableCreated
+    }
+    
+    set tableCreated(value) {
+        this._tableCreated = typeof value === 'undefined' ? false : !!value;
+    }
+
+    /**
+     * Creates a table in the database. Return null if the table already exists.
+     * 
+     * @param {DynamoTableOpts} dynamoTableOpts 
+     * @return {Promise} A promise that resolves with the result of creating the table.
+     */
+    async _createTable(dynamoTableOpts) {
+
+      const params = {
+        TableName: this.tableName,
+        AttributeDefinitions: [
+          {
+            AttributeName: 'key',
+            AttributeType: 'S'
+          }
+        ],
+        KeySchema: [
+          {
+            AttributeName: 'key',
+            KeyType: 'HASH'
+          }
+        ],
+        ProvisionedThroughput: {
+          ReadCapacityUnits: dynamoTableOpts?.readCapacityUnits || DynamoTableOpts.DEFAULT_READ_CAPACITY_UNITS,
```

Comment:
@Daniel-97 I am ok with dropping old versions  of Node.js support. The earliest version of Node.js supporting this `?.` operator is 14, isn't it?